src/monmove.cpp

```diff
@@ -1237,6 +1289,87 @@ bool monster::push_to( const tripoint &p, const int boost, const size_t depth )
     return true;
 }
 
+bool monster::take_item_at( const tripoint &p, std::function<std::list<item>::iterator(
+                                map_stack & )> selector )
+{
+    auto items = g->m.i_at( p );
+    auto to_pick_up = selector( items );
+
+    // TODO(sm): no volume/weight limits yet; AFAIK monsters/creatures have no notion of either
+    if( to_pick_up != items.end() ) {
+        // If there is a stack of items on the ground, only pick up as many of them as would
+        // count as "one" item for spawning purposes.
+        int default_charges = to_pick_up->type->charges_default();
+        if( default_charges > 1 && to_pick_up->charges > default_charges ) {
+            to_pick_up->mod_charges( -default_charges );
+            inv.push_back( *to_pick_up );
+            inv.back().charges = default_charges;
+
+            add_msg( m_warning, _( "The %1$s grabs %2$s!" ), name().c_str(),
+                     inv.back().display_name( default_charges ) );
+        } else {
+            std::move( to_pick_up, std::next( to_pick_up ), std::back_inserter( inv ) );
+            items.erase( to_pick_up );
+
+            add_msg( m_warning, _( "The %1$s grabs a %2$s!" ), name().c_str(),
```

Comment:
The message is also always shown, even when the monster is not visible.

Btw. why have two separate messages here? Note that this branch can be taken even when the item has more than 1 charge (e.g. default charges: 10, item charges: 5).

Reply:
I was originally trying to handle singular/plural cases, but evidently I didn't get it right (it's also not obvious whether, for instance, `hamburger (4)` is one burger that can be eaten 4 times, or many burgers). Also, as far as I can tell, CDDA doesn't have any facility for prefixing arbitrary strings "a"/"an" either. I'll keep it simple for now and just use one message.